app/models/test_track/analytics/mixpanel_client.rb

```diff
@@ -11,6 +11,14 @@ def mixpanel
       @mixpanel ||= Mixpanel::Tracker.new(ENV['MIXPANEL_TOKEN'])
     end
 
+    def event_name(assignment)
+      if assignment.feature_gate?
+        'FeatureGateExperienced'
+      else
+        'SplitAssigned'
+      end
+    end
```

Comment:
I could stuff this function into assignment as a method, e.g. `analytics_event_name`. It'd be your option whether you choose use it in your custom analytics plugin or use something else but you wouldn't have to duplicate the biz logic then.